source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
+that belong to an associated field. When specifying this option, {+odm+} stores
+an extra attribute on the associated models to store the count. Because of this,
+you must specify the ``Mongoid::Attributes::Dynamic`` module in the associated
+classes. The following example adds the ``counter_cache`` option to a ``Band``
+class and specifies the ``Mongoid::Attributes::Dynamic`` in a ``Label`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-counter-cache
+   :end-before: # end-counter-cache
+   :emphasize-lines: 4, 9
+
+Association Metadata
+--------------------
+
+When you define an association, {+odm+} stores metadata about that association.
+You can access the metadata by calling the ``reflect_on_association`` method on
+a model class or document, or by directly accessing the metadata on a specific
+document. The following example shows how to access metadata by using the
+``reflect_on_association`` method and by direct access:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-access-metadata
+   :end-before: # end-access-metadata
+
+.. note::
+
+   Replace ``<association_name>`` in the preceding example with the name of your
+   association.
+
+Attributes
+~~~~~~~~~~
+
+All associations contain attributes that store information about the associated
+document. Associations contain the following attributes:
+
+- ``_target``: Contains the proxied document or documents
+- ``_base``: Contains the document on which the association is defined
+- ``_association``: Contains information about the association
+
+The following example accesses each of the preceding attributes:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-attributes
+   :end-before: # end-attributes
+
+The following table shows the information stored in the ``_association``
+attribute:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 60
+
+   * - Method
+     - Description
+   * - ``Association#as``
+     - Returns the name of the parent to a polymorphic child.
+   * - ``Association#as?``
+     - Returns whether an ``as`` option exists.
+   * - ``Association#autobuilding?``
+     - Returns whether the association is autobuilding.
+   * - ``Association#autosaving?``
+     - Returns whether the association is autosaving.
+   * - ``Association#cascading_callbacks?``
+     - Returns whether the association has callbacks cascaded down from the parent.
+   * - ``Association#class_name``
+     - Returns the class name of the proxied document.
+   * - ``Association#cyclic?``
+     - Returns whether the association is a cyclic association.
+   * - ``Association#dependent``
+     - Returns the association's dependent option.
+   * - ``Association#destructive?``
+     - Returns ``true`` if the association has a dependent delete or destroy method.
```

Comment:
q: are 'delete' and 'destroy' names of specific methods, or just kinds of methods? if the former, should they be monospace?

Reply:
They're specific types of methods